### PR TITLE
feat(validation): season event-rate collapse check (catches the 2013 sack outage)

### PR DESCRIPTION
**Pull request by saiemgilani**
## Why

Row-level rules cannot see that an entire event type went **missing for a season** — every surviving row stays internally consistent, so nothing fires. That is exactly how the 2013 sack outage hid behind a fully green rule set.

## Root cause established — and it corrects my earlier read

The 2013 problem is **not a parser bug**:

| season | text says "sacked" | `sack` flag | `type.text == "Sack"` | rush-for-loss(≤−3)/game |
|---|---:|---:|---:|---:|
| 2012 | 3,254 | 3,254 | 3,047 | 2.86 |
| **2013** | **54** | **52** | **32** | **2.84** |
| 2014 | 3,601 | 3,537 | 3,213 | 3.06 |

The parser correctly flags 52 of the 54 plays whose text says "sacked". ESPN's own `type.text` classification has only 32 sacks that season. They were **not** folded into rushes either — rush-for-loss holds flat at 2.84/game, between 2012's 2.86 and 2014's 3.06.

**~3,000 sacks are simply absent from the 2013 feed. No parser change can recover data that is not in the source** — it needs a re-scrape or acceptance as a known-bad season.

## What this PR adds

`rate_anomaly` compares each season's per-game event rate against the median season. Calibration over **22 seasons × 12 flags = 264 season-flag cells**:

- **2013/sack = 0.015× median** (0.061/game vs 3.997/game) — the *only* cell below 0.5×
- next-worst cell anywhere = **0.560×** (fumble_vec, 2021)

The 0.5 floor sits in a wide empty gap between the real outage and normal era variation, so it is separated rather than tuned. Findings are WARN/`needs_judgment` — a rate move can be a genuine rule change (kickoff rates shift with touchback rules), so it is a triage signal, not an automatic defect. The floor is threshold-driven via `season_rate_floor`.

## Testing

169 contract tests pass (6 new). Run against the real 22-season dataset it emits exactly one finding — the 2013 sack collapse — with the measured numbers.

## Summary by Sourcery

Add a season-level event-rate collapse validation check and integrate it into the validation CLI for CFB play-by-play data.

New Features:
- Introduce a rate_anomaly check that detects seasons where key event rates collapse relative to the median season across historical data.

Enhancements:
- Wire the new rate_anomaly validation check into the validation CLI and leakage suites so it runs as part of standard validations.

Tests:
- Add contract tests covering normal variation, true collapses, threshold configuration, dataset/column mismatches, and edge cases like zero medians and short histories.

## Summary by CodeRabbit

- **New Features**
  - Added season-level rate anomaly validation for supported datasets.
  - Detects unusually low event rates compared with the median across seasons.
  - Reports affected seasons, event columns, expected and actual rates, and severity.
  - Supports configurable detection thresholds.

- **Bug Fixes**
  - Validation safely skips unsupported, incomplete, or insufficient data scenarios.
  - Handles zero or unavailable baselines without producing misleading findings.

### CODE REVIEW

File: tools/validation/checks/rate_anomaly.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    skipped -- there is no meaningful ratio to take.
+
+    Args:
+        dataset: Registered dataset name.
+        frame: The data frame under validation.
+        ctx: Check context supplying the domain and thresholds.
+
+    Returns:
+        One Finding per (flag, season) collapse; empty when every season holds.
+    """
```
[the remark this answers is not in the document]
Comment (saiemgilani) [reply tools/validation/checks/rate_anomaly.py:92]: Partially applied in 0d5e8662. Added the Example block (runnable, both default and custom-floor usage). Skipped a `Raises` section because `run` raises nothing by design and the guideline is to document *actual* error behavior: every degenerate input (unregistered dataset, missing columns, <3 seasons, zero-median flag) short-circuits to an empty list, since a check that aborts the run is worse than one that reports nothing. That no-raise contract is now stated explicitly under `Returns` so it is documented rather than merely absent.
